src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,204 @@
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+
+import duke.exception.IllegalCommandException;
+import duke.exception.EmptyCommandException;
+import duke.task.Task;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+public class Duke {
+
+    public static ArrayList<Task> tasks = new ArrayList<>();
+    public static final String FILE_NAME = "duke.txt";
+    public static final String DIR_NAME = "data";
+    public static final String FILE_PATH = DIR_NAME + "/" + FILE_NAME;
+
+    /**
+     * Prints a farewell message when the user exits the program.
+     */
+    public static void commandBye() {
+        System.out.println("Bye! I hope to see you again soon :)");
+    }
+
+    /**
+     * Lists all tasks with their status.
+     */
+    public static void commandList() {
+        System.out.println("Here are your current tasks and their status:");
+        for (Task task : tasks) {
+            System.out.println((tasks.indexOf(task) + 1) + ". " + task);
+        }
+    }
+
+    /**
+     * Adds a task of a certain type defined by the user to the list.
+     *
+     * @param line The command entered by the user.
+     */
+    public static void addTask(String line) throws EmptyCommandException, IllegalCommandException, IOException {
+
+        if (!line.startsWith("todo") && !line.startsWith("deadline") && !line.startsWith("event")) {
+            throw new IllegalCommandException();
+        }
+
+        if (line.split(" ").length < 2) {
+            throw new EmptyCommandException();
+        }
+
+        if (line.startsWith("todo")) {
+            tasks.add(Task.getTaskCount(), new Todo(line.replace("todo ", "")));
+        } else if (!line.contains("/")) {
+            throw new IllegalCommandException();
+        } else if (line.startsWith("deadline")) {
+            String[] words = line.split(" /by ");
+            tasks.add(Task.getTaskCount(), new Deadline(words[0].replace("deadline ", ""), words[1]));
+        } else if (line.startsWith("event")) {
+            String[] words = line.split(" /at ");
+            tasks.add(Task.getTaskCount(), new Event(words[0].replace("event ", ""), words[1]));
+        }
+
+        System.out.println("I've added that to your list:\n" + tasks.get(Task.getTaskCount() - 1));
+        System.out.println("Now you have " + Task.getTaskCount() + " tasks in the list.");
+        saveData();
+    }
+
+    /**
+     * Sets a task in the list (specified by the user) as done.
+     * The command is entered in the format "done X", where X is
+     * the task's number.
+     *
+     * @param line The command entered by the user.
+     */
+    public static void doneTask(String line) throws EmptyCommandException, IllegalCommandException, IOException {
+        String[] words = line.split(" ");
+        if (words.length < 2) {
```

Comment:
You can make the [happy path](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent) more prominent here. 
Eg. if `words.length < 2`, you will throw an exception already, so you don't need to nest the rest of the code in `else`